src/ImageSharp/Common/Helpers/Vector4Utilities.cs

```diff
@@ -56,13 +64,42 @@ public static void UnPremultiply(ref Vector4 source)
         [MethodImpl(InliningOptions.ShortMethod)]
         public static void Premultiply(Span<Vector4> vectors)
         {
-            // TODO: This method can be AVX2 optimized using Vector<float>
-            ref Vector4 baseRef = ref MemoryMarshal.GetReference(vectors);
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Avx2.IsSupported && vectors.Length >= 2)
+            {
+                ref Vector256<float> vectorsBase =
+                    ref Unsafe.As<Vector4, Vector256<float>>(ref MemoryMarshal.GetReference(vectors));
 
-            for (int i = 0; i < vectors.Length; i++)
+                Vector256<int> mask =
+                    Unsafe.As<byte, Vector256<int>>(ref MemoryMarshal.GetReference(PermuteAlphaMask8x32));
+
+                // Divide by 2 as 4 elements per Vector4 and 8 per Vector256<float>
+                ref Vector256<float> vectorsLast = ref Unsafe.Add(ref vectorsBase, (IntPtr)((uint)vectors.Length / 2u));
+
+                while (Unsafe.IsAddressLessThan(ref vectorsBase, ref vectorsLast))
+                {
+                    Vector256<float> source = vectorsBase;
+                    Vector256<float> multiply = Avx2.PermuteVar8x32(source, mask);
```

Comment:
You actually don't need to permute here since you're not crossing 128-bit lanes. `Avx.Shuffle(source, source, 0b_11_11_11_11)` will do the same thing with lower latency while eliminating the need to load the mask register.

Reply:
Thanks @saucecontrol I didn't know Shuffle had that overload. Slight speedup. 

|              Method |     Mean |    Error |   StdDev | Ratio | Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------------------- |---------:|---------:|---------:|------:|------:|------:|------:|----------:|
| PremultiplyBaseline | 37.64 us | 1.482 us | 0.081 us |  1.00 |     - |     - |     - |         - |
|         Premultiply | 27.42 us | 1.738 us | 0.095 us |  0.73 |     - |     - |     - |         - |

|                Method |      Mean |     Error |    StdDev | Ratio | Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------------------- |----------:|----------:|----------:|------:|------:|------:|------:|----------:|
| UnPremultiplyBaseline | 37.753 us | 3.9513 us | 0.2166 us |  1.00 |     - |     - |     - |         - |
|         UnPremultiply |  1.322 us | 0.0998 us | 0.0055 us |  0.04 |     - |     - |     - |         - |

Reply:
Still don't understand the same method with `Divide` instead of `Multiply` results in a ~30X difference in benchmark result though 😖

Reply:
Ha, I didn't read through all the comments and missed that bit.  Your baseline UnPremultiply method in the benchmark is multiplying instead of dividing, but I don't see why the vectorized version is coming out so much faster.  Will have a look after sleep if you don't figure it out ;)

Reply:
I knew it! I knew there was a mistake! Thanks!

I guess my computer just doesn't like multiplying stuff. The baseline is way faster now too. 

|                Method |     Mean |     Error |    StdDev | Ratio | Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------------------- |---------:|----------:|----------:|------:|------:|------:|------:|----------:|
| UnPremultiplyBaseline | 2.018 us | 0.1879 us | 0.0103 us |  1.00 |     - |     - |     - |         - |
|         UnPremultiply | 1.255 us | 0.0452 us | 0.0025 us |  0.62 |     - |     - |     - |         - |

Reply:
That's strange the division is showing lower times.  Is that just an iteration count difference between the Premultiply and UnPremultiply runs?  BDN is too clever sometimes.

Reply:
Nah... Exactly the same setup. I think it's optimizing something away.